files/default/lib/bumpzone.rb

```diff
@@ -0,0 +1,112 @@
+#!/opt/chef/embedded/bin/ruby
+require 'git'
+require 'json'
+require 'octokit'
+
+# Library to bump DNS zones
+class BumpZone
+  @commit_msg = "\n\n"
+  @warning_msg = "Warning:\n\n"
+  @warning = false
+  ONELINE_REGEX = /(^[\w\d\.\-]+\s+[\d]+\s+IN\s+SOA.*[\w\d\.\-]+\s+)(\d{10})(\s+[\d]+\s+[\d]+\s+[\d]+\s+[\d]+)/
+  MULTILINE_REGEX =
+    /(^[\w\d\.\-]+\s+IN\s+SOA\s+[\w\d\.\-]+\s+[\w\d\.\-]+\s+\(\s?\n\s+)(\d{10})(.*\n.*\n.*\n.*\n.*\n\s+\))/
+
+  class << self
+    attr_reader :commit_msg
+  end
+
+  def self.add_commit_msg(msg)
+    @commit_msg << msg
+  end
+
+  def self.add_warning_msg(msg)
+    @warning = true
+    @warning_msg << msg
+  end
+
+  def self.regex(file, regex)
+    !(file !~ regex)
+  end
+
+  def self.soa_oneline?(zone)
+    if regex(zone, ONELINE_REGEX)
+      true
+    elsif regex(zone, MULTILINE_REGEX)
+      false
```

Comment:
Either way is fine IMO. If you change the purpose, you may wish to rename it, and if you change the return type you'd want to remove the question mark. But it's probably best not to spend too much time picking colors on this particular bikeshed.